eng/templates/stages/retain-build.yml

```diff
@@ -0,0 +1,23 @@
+stages:
+- stage: RetainProductionBuild
+  displayName: Retain Production Build
+  dependsOn: []
+  jobs:
+  - job: RetainBuild
+    displayName: Retain Production Build
+    pool:
+      name: NetCore1ESPool-Internal-NoMSI
+      demands: ImageOverride -equals 1es-windows-2022
+
+    variables:
+    - name: AzdoOrgUri
+      value: https://dev.azure.com/dnceng
+    - name: AzdoProject
+      value: internal
```

Comment:
Is there value in defining these?

Reply:
Currently, not really, but if we add more things to this job, it might be nice to have all variables defined at the beginning of it. I guess it's a style thing. I don't mind changing it

Reply:
This file is quite small so to me it just felt like it has grown for no actual reason. If this step would be injected in a larger file, I'd be up for defining variables on top. But I will leave this up to you.